/*
 * This file was automatically generated by EvoSuite
 * Tue Sep 05 19:27:20 GMT 2017
 */

package org.araguacaima.exception.core;

import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.Test;
import org.junit.runner.RunWith;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

@RunWith(EvoRunner.class)
@EvoRunnerParameters(mockJVMNonDeterminism = true,
                     useVFS = true,
                     useVNET = true,
                     resetStaticState = true,
                     separateClassLoader = true,
                     useJEE = true)
public class ApplicationException_ESTest extends ApplicationException_ESTest_scaffolding {

    @Test(timeout = 4000)
    public void test0()
            throws Throwable {
        ApplicationException applicationException0 = new ApplicationException("<Ux*oJ21v,l 8p<");
        applicationException0.perform();
        assertEquals("<Ux*oJ21v,l 8p<", applicationException0.getKeyCode());
    }

    @Test(timeout = 4000)
    public void test1()
            throws Throwable {
        Severity severity0 = Severity.ERROR;
        ApplicationException applicationException0 = new ApplicationException("-J:MIF6}1Q", severity0, "P", severity0);
        assertEquals("-J:MIF6}1Q", applicationException0.getKeyCode());
    }

    @Test(timeout = 4000)
    public void test2()
            throws Throwable {
        Severity severity0 = Severity.DEBUG;
        ApplicationException applicationException0 = new ApplicationException((String) null, severity0, (String) null);
        assertNull(applicationException0.getKeyCode());
    }

    @Test(timeout = 4000)
    public void test3()
            throws Throwable {
        ApplicationException applicationException0 = new ApplicationException("", (Exception) null);
        assertEquals("", applicationException0.getKeyCode());
    }

    @Test(timeout = 4000)
    public void test4()
            throws Throwable {
        Object[] objectArray0 = new Object[2];
        ApplicationException applicationException0 = new ApplicationException("", (Severity) null, objectArray0);
        assertEquals("", applicationException0.getExtendedMessage());
    }

    @Test(timeout = 4000)
    public void test5()
            throws Throwable {
        Severity severity0 = Severity.DEBUG;
        ApplicationException applicationException0 = new ApplicationException("", severity0);
        assertEquals("", applicationException0.getExtendedMessage());
    }

    @Test(timeout = 4000)
    public void test6()
            throws Throwable {
        ApplicationException applicationException0 = new ApplicationException("DEBUG",
                (Severity) null,
                (Object) "DEBUG");
        assertEquals("", applicationException0.getExtendedMessage());
    }
}
